sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java

```diff
@@ -69,4 +72,52 @@ public void start_StartedThenComplete() throws InterruptedException {
         Assert.assertTrue(listener.finishCalled);
         Assert.assertFalse(listener.errorCalled);
     }
+
+    @Test
+    public void testRequestFilters() throws Exception {
+
+        //to test the filters we make an invalid request
+
+        TestReplicationListener listener = new TestReplicationListener();
+        PullReplication pullReplication = createPullReplication();
+        pullReplication.requestFilters.add(new InvalidJSONFilter());
+        Replicator replicator = ReplicatorFactory.oneway(pullReplication);
+        replicator.getEventBus().register(listener);
+        replicator.start();
+
+        while(replicator.getState() != Replicator.State.COMPLETE && replicator.getState() != Replicator.State.ERROR) {
+            Thread.sleep(1000);
```

Comment:
I was following the lead of the other tests which wait for replications to complete, we can probably lower the wait time, I'll investigate